internal/storage/webhook/webhook.go

```diff
@@ -0,0 +1,169 @@
+package webhookstorage
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/darchlabs/synchronizer-v2/internal/storage"
+	"github.com/darchlabs/synchronizer-v2/pkg/webhook"
+	"github.com/pkg/errors"
+)
+
+type Storage struct {
+	storage *storage.S
+}
+
+func New(s *storage.S) *Storage {
+	return &Storage{
+		storage: s,
+	}
+}
+
+func (s *Storage) CreateWebhook(wh *webhook.Webhook) (*webhook.Webhook, error) {
+	query := `
+		INSERT INTO webhooks (id, entity_type, entity_id, endpoint, payload, created_at, updated_at, sent_at, next_retry_at) 
+		VALUES (:id, :entity_type, :entity_id, :endpoint, :payload, :created_at, :updated_at, :sent_at, :next_retry_at)
+		RETURNING id
+	`
+
+	rows, err := s.storage.DB.NamedQuery(query, wh)
+	if err != nil {
+		return nil, errors.Wrap(err, "webhookstorage: error creating webhook")
+	}
+	defer rows.Close()
+
+	if rows.Next() {
+		var id string
+		if err := rows.Scan(&id); err != nil {
+			return nil, errors.Wrap(err, "webhookstorage: error scanning webhook ID")
+		}
+		wh.ID = id
+	} else {
+		return nil, errors.New("webhookstorage: no ID returned after webhook creation")
+	}
+
+	createdWebhook, err := s.GetWebhookByID(wh.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return createdWebhook, nil
+}
+
+func (s *Storage) UpdateWebhook(wh *webhook.Webhook) (*webhook.Webhook, error) {
+	query := `
+		UPDATE webhooks 
+		SET endpoint = :endpoint, payload = :payload, status = :status, attempts = :attempts, next_retry_at = :next_retry_at, updated_at = :updated_at, sent_at = :sent_at 
+		WHERE id = :id
+	`
+
+	_, err := s.storage.DB.NamedExec(query, wh)
+	if err != nil {
+		return nil, errors.Wrap(err, "webhookstorage: error updating webhook")
+	}
+
+	updatedWebhook, err := s.GetWebhookByID(wh.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return updatedWebhook, nil
+}
```

Comment:
here, we could create a special input for the update query with pointers on it so we could leverage the `COEALESCE` method in sql. Here is an example of how to use it https://github.com/darchlabs/nodes/blob/master/internal/storage/instance/update_query.go